lib/ncr/work_order_reapproval_checker.rb

```diff
@@ -0,0 +1,46 @@
+module Ncr
+  class WorkOrderReapprovalChecker
+    attr_reader :work_order
+
+    def initialize(work_order)
+      @work_order = work_order
+    end
+
+    def current_user
+      work_order.modifier
+    end
+
+    def amount_increased?
+      changes = work_order.previous_changes
+      changes.key?('amount') && (work_order.amount > changes['amount'].first)
+    end
+
+    def protected_fields_changed?
+      changes = work_order.previous_changes
+      protected_fields.any? do |field|
+        values = changes[field.to_s]
+        values && values[0].present?
+      end
+    end
+
+    def budget_approver?
+      work_order.budget_approvers.include?(current_user)
+    end
+
+    def requires_budget_reapproval?
+      work_order.approved? && (
+        self.amount_increased? || (
+          self.protected_fields_changed? &&
+          !self.budget_approver?
+        )
+      )
+    end
```

Comment:
@phirefly This is the important part, in terms of determining if a request needs to be re-approved or not.